script/tool/test/xctest_command_test.dart

```diff
@@ -123,13 +88,93 @@ void main() {
       );
     });
 
+    test('allows target filtering', () async {
+      final Directory pluginDirectory1 = createFakePlugin('plugin', packagesDir,
+          platformSupport: <String, PlatformSupport>{
+            kPlatformMacos: PlatformSupport.inline,
+          });
+
+      final Directory pluginExampleDirectory =
+          pluginDirectory1.childDirectory('example');
+
+      final List<String> output = await runCapturingPrint(runner, <String>[
+        'xctest',
+        '--macos',
+        '--test-target=RunnerTests',
+      ]);
+
+      expect(
+          output,
+          contains(
+              contains('Successfully ran macOS xctest for plugin/example')));
+
+      expect(
+          processRunner.recordedCalls,
+          orderedEquals(<ProcessCall>[
+            ProcessCall(
+                'xcrun',
+                const <String>[
+                  'xcodebuild',
+                  'test',
+                  '-workspace',
+                  'macos/Runner.xcworkspace',
+                  '-scheme',
+                  'Runner',
+                  '-configuration',
+                  'Debug',
+                  '-only-testing:RunnerTests',
+                  'GCC_TREAT_WARNINGS_AS_ERRORS=YES',
+                ],
+                pluginExampleDirectory.path),
+          ]));
+    });
+
+    test('reports skips with no tests', () async {
+      final Directory pluginDirectory1 = createFakePlugin('plugin', packagesDir,
+          platformSupport: <String, PlatformSupport>{
+            kPlatformMacos: PlatformSupport.inline,
+          });
+
+      final Directory pluginExampleDirectory =
+          pluginDirectory1.childDirectory('example');
+
+      // Exit code 66 from testing indicates no tests.
+      final MockProcess noTestsProcessResult = MockProcess();
+      noTestsProcessResult.exitCodeCompleter.complete(66);
+      processRunner.mockProcessesForExecutable['xcrun'] = <io.Process>[
+        noTestsProcessResult,
+      ];
+      final List<String> output =
+          await runCapturingPrint(runner, <String>['xctest', '--macos']);
+
+      expect(output, contains(contains('No tests found.')));
+
+      expect(
+          processRunner.recordedCalls,
+          orderedEquals(<ProcessCall>[
+            ProcessCall(
+                'xcrun',
+                const <String>[
+                  'xcodebuild',
+                  'test',
+                  '-workspace',
+                  'macos/Runner.xcworkspace',
+                  '-scheme',
+                  'Runner',
+                  '-configuration',
+                  'Debug',
+                  'GCC_TREAT_WARNINGS_AS_ERRORS=YES',
+                ],
+                pluginExampleDirectory.path),
+          ]));
+    });
+
     group('iOS', () {
       test('skip if iOS is not supported', () async {
-        createFakePlugin('plugin', packagesDir, extraFiles: <String>[
-          'example/test',
```

Comment:
Opportunistic cleanup; this `extraFiles` was copy/pasted from some initial test into lots of other tests that don't actually need it, and I've been cleaning it up as I notice.